groupnorm: eager mode QAT support

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39413 graph mode: util for fusion of functions which require observation
* #39203 quant layer/group/instance norm: make weights and biases optional
* #39096 instancenorm: static quant graph mode support
* #39095 groupnorm: graph mode static quant support
* #39094 layernorm: eager mode qat support
* #39093 instancenorm: eager mode QAT support
* **#39092 groupnorm: eager mode QAT support**
* #39091 instancenorm: eager static quant support
* #39090 groupnorm: eager static quant support
* #37802 quant: clean up normalization channels_last handling

Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885261](https://our.internmc.facebook.com/intern/diff/D21885261)

vkuzo committed:
groupnorm: eager mode QAT support

Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantize.py:
```diff
@@ -912,6 +913,33 @@ def checkQuantized(model):
                 checkQuantized(model)
 
+    def test_normalization(self):
+        for qengine in supported_qengines:
+            with override_quantized_engine(qengine):
```
replace with `@override_qengines`?

supriyar reviewed (approved):
looks good, left a small comment.

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885261](https://our.internmc.facebook.com/intern/diff/D21885261)

[ghstack-poisoned]

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885261](https://our.internmc.facebook.com/intern/diff/D21885261)

[ghstack-poisoned]

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885261](https://our.internmc.facebook.com/intern/diff/D21885261)

[ghstack-poisoned]

vkuzo committed:
Update on "groupnorm: eager mode QAT support"


Summary:

Adds eager mode QAT support for GroupNorm.

Test Plan:

```
python test/test_quantization.py TestQuantizationAwareTraining.test_normalization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885261](https://our.internmc.facebook.com/intern/diff/D21885261)

[ghstack-poisoned]
